thundra/plugins/metric/metric_plugin.py

```diff
@@ -21,92 +22,116 @@ def __init__(self):
         self.system_cpu_total_end = float(0)
         self.process_cpu_usage_start = float(0)
         self.process_cpu_usage_end = float(0)
-        self.stat_data = {}
+        self.metric_data = {}
         self.reporter = {}
+        self.tracer = ThundraTracer()
 
     def before_invocation(self, data):
         self.reporter = data['reporter']
         context = data['context']
-        stat_time = time.time() * 1000
-        self.stat_data = {
-            'transactionId': data['transactionId'],
-            'applicationName': getattr(context, 'function_name', None),
+        metric_time = time.time() * 1000
+        function_name = getattr(context, constants.CONTEXT_FUNCTION_NAME, None)
+
+        active_span = self.tracer.get_active_span()
+
+        self.metric_data = {
+
+            'type': "Metric",
+            'agentVersion': '',
```

Comment:
Even though this can't be detected automatically, we can get this from constants and the constant is updated manually for every release. Can start with `2.0.0` for the first release with new data model